Mathlib/Topology/Category/Stonean/Basic.lean

```diff
@@ -143,4 +142,80 @@ instance : Faithful toProfinite := {}
 example : toProfinite ⋙ profiniteToCompHaus = toCompHaus :=
   rfl
 
+/-- Every Stonean space is projective in `CompHaus` -/
+instance (X : Stonean) : Projective X.compHaus where
+  factors := by
+    intro B C φ f _
+    haveI : ExtremallyDisconnected X.compHaus.toTop := X.extrDisc
+    have hf : f.1.Surjective
+    · rwa [CompHaus.epi_iff_surjective] at *
+    obtain ⟨f', h⟩ := CompactT2.ExtremallyDisconnected.projective φ.continuous f.continuous hf
+    use ⟨f', h.left⟩
+    ext
+    exact congr_fun h.right _
+
 end Stonean
+
+namespace CompHaus
+
+/-- If `X` is compact Hausdorff, `presentation X` is an extremally disconnected space
+  equipped with an epimorphism down to `X`. It is a "constructive" witness to the
+  fact that `CompHaus` has enough projectives.  -/
+noncomputable
+def presentation (X : CompHaus) : Stonean where
+  compHaus := (projectivePresentation X).p
+  extrDisc := by
+    refine' CompactT2.Projective.extremallyDisconnected
+      (@fun Y Z _ _ _ _ _ _ f g hfcont hgcont hgsurj => _)
+    let g₁ : (CompHaus.of Y) ⟶ (CompHaus.of Z) := ⟨g, hgcont⟩
+    let f₁ : (projectivePresentation X).p ⟶ (CompHaus.of Z) := ⟨f, hfcont⟩
+    have hg₁ : Epi g₁ := (epi_iff_surjective _).2 hgsurj
+    refine' ⟨Projective.factorThru f₁ g₁, (Projective.factorThru f₁ g₁).2, funext (fun _ => _)⟩
+    change (Projective.factorThru f₁ g₁ ≫ g₁) _ = f _
+    rw [Projective.factorThru_comp]
+    rfl
+
+/-- The morphism from `presentation X` to `X`. -/
+noncomputable
+def presentationπ (X : CompHaus) : X.presentation.compHaus ⟶ X :=
+  (projectivePresentation X).f
+
+/-- The morphism from `presentation X` to `X` is an epimorphism. -/
+noncomputable
+instance epiPresentπ (X : CompHaus) : Epi X.presentationπ :=
```

Comment:
If you agree with the above suggestion, then this might become
```suggestion
instance presentation.epi_π (X : CompHaus) : Epi X.presentationπ :=
```